### PR TITLE
Improve client stderr logs

### LINKED ISSUE #590: Improve devnet logs
- [x] Shorten hash to 4 symbols "f4gd"
- [x] Expand blocks message from just hash to number of TX. Block 141 "gdfs", 5 tx, 3 new tx
- [x] Print every transaction including status
- [x] Convert vec<u8> arrays to preview. Like [10, 205, 20, 30, ... (104 more)]
- [ ] Unicorns

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Fixes #590 
- Adding logging with the details of produced blocks.
- Improving Debug fmt using padded structure. We can write a blog post on it. It's that cool.
- Adding args, method_names and wasm_code and other avoided items to the logs
- Truncating long hashes, (e.g. signatures)

Log examples below:

```
[2019-02-26T02:16:33Z DEBUG node_http::api] Received transaction SignedTransaction {
    body: DeployContract(
        DeployContractTransaction {
            nonce: 2,
            contract_id: test1235,
            wasm_byte_array: (27463)[0, 97, … 0, 121]
        }
    ),
    signature: (88)`4Fmo82…p1RozA`,
    hash: `71QBr5Z1UdWb3B9kAtzAdzfaZALrNqHm8z4cqkv9ozLb`
}
[2019-02-26T02:16:33Z INFO  client] Producing block index: 49, beacon hash = `CiJ2HLj7hHZa8c3FzDyGPum6q274WvriwmKWmak9F2ma`, shard hash = `9j2vS1NDGFSQMsiWPebAhWMG5w6oDY8S2mRR22DCZzUi`
[2019-02-26T02:16:33Z DEBUG client] Input Transactions: [SignedTransaction {
    body: DeployContract(
        DeployContractTransaction {
            nonce: 2,
            contract_id: test1235,
            wasm_byte_array: (27463)[0, 97, … 0, 121]
        }
    ),
    signature: (88)`4Fmo82…p1RozA`,
    hash: `71QBr5Z1UdWb3B9kAtzAdzfaZALrNqHm8z4cqkv9ozLb`
} -> TransactionResult {
    status: Completed,
    logs: [],
    receipts: []
}]
[2019-02-26T02:16:33Z DEBUG client] Output Transactions: []

[2019-02-26T02:10:38Z DEBUG node_http::api] Call view function test1235.generateLogs(`{}`)
[2019-02-26T02:10:40Z DEBUG node_runtime::state_viewer] (exec time 476.40ms) result of execution: ExecutionOutcome {
    gas_used: 2859,
    mana_used: 0,
    mana_left: 0,
    return_data: Ok(
        None
    ),
    balance: 0,
    random_seed: `8J3hnhrL5ik1BwLn33rXVYpPf12VtAXYfneJLZ787DqQ`,
    logs: [
        "LOG: log1",
        "LOG: log2"
    ]
}
[2019-02-26T02:10:46Z DEBUG node_http::api] Call view function test1235.benchmark_storage(`{"n":1000}`)
[2019-02-26T02:10:47Z DEBUG node_runtime::state_viewer] (exec time 78.55ms) result of execution: ExecutionOutcome {
    gas_used: 4296936,
    mana_used: 0,
    mana_left: 0,
    return_data: Ok(
        Value(
            `{"result":"499500"}`
        )
    ),
    balance: 0,
    random_seed: `8J3hnhrL5ik1BwLn33rXVYpPf12VtAXYfneJLZ787DqQ`,
    logs: []
}
[2019-02-26T02:10:54Z DEBUG node_http::api] Call view function test1235.getValue(`{}`)
[2019-02-26T02:10:54Z DEBUG node_runtime::state_viewer] (exec time 13.63ms) result of execution: ExecutionOutcome {
    gas_used: 11199,
    mana_used: 0,
    mana_left: 0,
    return_data: Ok(
        Value(
            `{"result":"hi"}`
        )
    ),
    balance: 0,
    random_seed: `8J3hnhrL5ik1BwLn33rXVYpPf12VtAXYfneJLZ787DqQ`,
    logs: []
}
[2019-02-26T02:11:14Z DEBUG node_http::api] View account alice.near
[2019-02-26T02:11:14Z DEBUG node_http::api] Received transaction SignedTransaction {
    body: FunctionCall(
        FunctionCallTransaction {
            nonce: 16,
            originator: alice.near,
            contract_id: test1235,
            method_name: `generateLogs3123`,
            args: `{}`,
            amount: 0
        }
    ),
    signature: (88)`5LZP7H…wYnpqU`,
    hash: `7CuBfiuAjZ46WtbrmF4B2cxjVRZwzqmZ4HgPNB7eCUCd`
}
[2019-02-26T02:11:14Z INFO  client] Producing block index: 46, beacon hash = `DtMnBKrG4q2vQGchwKRLBVEMccaLfCGYRebcNusq3zyB`, shard hash = `948LqxSjxKm8ugfdwALDEZNBof1GMD2rqKSYxiSvitGP`
[2019-02-26T02:11:14Z DEBUG client] Input Transactions: [SignedTransaction {
    body: FunctionCall(
        FunctionCallTransaction {
            nonce: 16,
            originator: alice.near,
            contract_id: test1235,
            method_name: `generateLogs3123`,
            args: `{}`,
            amount: 0
        }
    ),
    signature: (88)`5LZP7H…wYnpqU`,
    hash: `7CuBfiuAjZ46WtbrmF4B2cxjVRZwzqmZ4HgPNB7eCUCd`
} -> TransactionResult {
    status: Completed,
    logs: [],
    receipts: [
        `EFHEu6bdCRNzM6Y9kMg64RtatJfVuoqmi17xiTnctErZ`
    ]
}]
[2019-02-26T02:11:14Z DEBUG client] Output Transactions: [
    ReceiptTransaction {
        originator: "alice.near",
        receiver: "test1235",
        nonce: `EFHEu6bdCRNzM6Y9kMg64RtatJfVuoqmi17xiTnctErZ`,
        body: NewCall(
            AsyncCall {
                amount: 0,
                mana: 19,
                method_name: `generateLogs3123`,
                args: `{}`,
                callback: None,
                accounting_info: AccountingInfo {
                    originator: "alice.near",
                    contract_id: None
                }
            }
        )
    }
]
[2019-02-26T02:11:15Z DEBUG node_runtime] Runtime error: wasm async call execution failed with error: Wasmer(CallError(Resolve(ExportNotFound { name: "near_func_generateLogs3123" })))
[2019-02-26T02:11:15Z INFO  client] Producing block index: 47, beacon hash = `J5rUeG4WMYjEFEbr1sQmmY7E1YydJjvcgQD2Jwty45v2`, shard hash = `AGKLTTo4c66t2hY5bF1Wup541FXDU2r4N4y31vwU8zxn`
[2019-02-26T02:11:15Z DEBUG client] Input Transactions: [ReceiptTransaction {
    originator: "alice.near",
    receiver: "test1235",
    nonce: `EFHEu6bdCRNzM6Y9kMg64RtatJfVuoqmi17xiTnctErZ`,
    body: NewCall(
        AsyncCall {
            amount: 0,
            mana: 19,
            method_name: `generateLogs3123`,
            args: `{}`,
            callback: None,
            accounting_info: AccountingInfo {
                originator: "alice.near",
                contract_id: None
            }
        }
    )
} -> TransactionResult {
    status: Failed,
    logs: [
        "Runtime error: wasm async call execution failed with error: Wasmer(CallError(Resolve(ExportNotFound { name: \"near_func_generateLogs3123\" })))"
    ],
    receipts: [
        `9jXc6nVHXBbLvR2DDR1RWPXoqKzukSNwrwU1HgBSSAQC`
    ]
}]
[2019-02-26T02:11:15Z DEBUG client] Output Transactions: [
    ReceiptTransaction {
        originator: "test1235",
        receiver: "alice.near",
        nonce: `9jXc6nVHXBbLvR2DDR1RWPXoqKzukSNwrwU1HgBSSAQC`,
        body: ManaAccounting(
            ManaAccounting {
                accounting_info: AccountingInfo {
                    originator: "alice.near",
                    contract_id: None
                },
                mana_refund: 19,
                gas_used: 0
            }
        )
    }
]
[2019-02-26T02:11:15Z INFO  client] Producing block index: 48, beacon hash = `3Ti5Jz2oi5L9u55h5d3eTvxumTrouxPPj7dZ8JHC6yzD`, shard hash = `B2ruFozfc2Z9LcYZ1xxRgcHbbDCvUN7Y8JCPFSQYFdVu`
[2019-02-26T02:11:15Z DEBUG client] Input Transactions: [ReceiptTransaction {
    originator: "test1235",
    receiver: "alice.near",
    nonce: `9jXc6nVHXBbLvR2DDR1RWPXoqKzukSNwrwU1HgBSSAQC`,
    body: ManaAccounting(
        ManaAccounting {
            accounting_info: AccountingInfo {
                originator: "alice.near",
                contract_id: None
            },
            mana_refund: 19,
            gas_used: 0
        }
    )
} -> TransactionResult {
    status: Completed,
    logs: [],
    receipts: []
}]
[2019-02-26T02:11:15Z DEBUG client] Output Transactions: []
```

### CODE REVIEW

File: node/client/src/lib.rs [hunk 1]
```diff
@@ -223,11 +223,26 @@ impl Client {
              This should never happen, because block production is atomic."
         );
 
+        info!(target: "client", "Producing block index: {:?}, beacon hash = {:?}, shard hash = {:?}",
+            block.body.header.index,
+            block.hash,
+            shard_block.hash,
+        );
+        let block_receipts = get_all_receipts(shard_block.body.receipts.iter());
```
Comment (nearmax) [inline node/client/src/lib.rs]: Wrap this code into https://docs.rs/log/0.4.6/log/macro.log_enabled.html

**Comment by vgrichina** [comment]
@evgenykuzyakov  logs look really nice! However can we also have an option to flatten output into single line? Hard to grep logs if message spans more than one line.

File: node/client/src/lib.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply node/client/src/lib.rs]: Good point

**Comment by evgenykuzyakov** [comment]
> logs look really nice! However can we also have an option to flatten output into single line? Hard to grep logs if message spans more than one line.

I don't think there is a single line option. We would have to update all "{:#?}" at compile time to "{:?}" for this to work. Or I guess only high level use, since it would flatten most.

**Pull request merged**